calico_containers/calico_ctl/container.py

```diff
@@ -443,11 +477,44 @@ def container_ip_remove(container_id, ip, interface):
         print "Error updating networking in container. Aborting."
         sys.exit(1)
 
-    client.unassign_address(pool, address)
+    client.release_ips({address})
 
     print "IP %s removed from %s" % (ip, container_id)
 
 
+def get_ip_and_pool(ip):
+    if ip.lower() in ("ipv4", "ipv6"):
+        if '4' in ip:
+            result = assign_any(1, 0)
+            ip = result[0][0]
+        elif '6' in ip:
+            result = assign_any(0, 1)
+            ip = result[1][0]
+        pool = get_pool_or_exit(ip)
+    elif ip is not None and '/' in ip:
+        requested_pool = IPNetwork(ip)
+        requested_version = requested_pool.version
+        pool = get_pool_or_exit(requested_pool.ip)
```

Comment:
We've already confirmed the pool exists in `validate_arguments()`.  You can just create the pool object directly
```
pool = IPPool(ip)
```

An `IPPool` subclasses `IPNetwork` so you can then do `pool.version` and the like.

This saves a few redundant reads of the datastore.